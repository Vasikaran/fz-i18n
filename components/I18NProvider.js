import React,{PropTypes, Children} from 'react';

const emptyObj={};
export class I18NProvider extends React.Component{
	getChildContext() {
    return {i18n: this.props.i18n};
  }
  render(){
  	return Children.only(this.props.children)
  }

}

I18NProvider.defaultProps = {
	i18n:emptyObj
}
I18NProvider.childContextTypes = {
  i18n: PropTypes.object
};

export default I18N;