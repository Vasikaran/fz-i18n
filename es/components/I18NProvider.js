import _typeof from 'babel-runtime/helpers/typeof';
import _Promise from 'babel-runtime/core-js/promise';
import _Object$getPrototypeOf from 'babel-runtime/core-js/object/get-prototype-of';
import _classCallCheck from 'babel-runtime/helpers/classCallCheck';
import _createClass from 'babel-runtime/helpers/createClass';
import _possibleConstructorReturn from 'babel-runtime/helpers/possibleConstructorReturn';
import _inherits from 'babel-runtime/helpers/inherits';
import React, { Children } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { getI18NValue, userDateFormat } from '../utils';

var emptyObj = {};
var dummy = function dummy(key, values) {
  console.log('Not Mount <I18NProvider/> component.');
  return key;
};
export var i18NProviderUtils = {
  getI18NValue: dummy,
  userDateFormat: dummy
};

var I18NProvider = function (_React$Component) {
  _inherits(I18NProvider, _React$Component);

  function I18NProvider(props, context) {
    _classCallCheck(this, I18NProvider);

    var _this = _possibleConstructorReturn(this, (I18NProvider.__proto__ || _Object$getPrototypeOf(I18NProvider)).call(this, props, context));

    _this.i18n = props.i18n;
    _this.timeZone = props.timeZone;
    _this.user = props.user;
    _this.permission = props.permission;
    _this.license = props.license;
    _this.promise = null;
    _this.resolve = null;
    _this.reject = null;
    _this.state = { reRender: false };
    i18NProviderUtils.getI18NValue = getI18NValue(props.i18n);
    i18NProviderUtils.userDateFormat = userDateFormat(i18NProviderUtils.getI18NValue, props.timeZone);
    return _this;
  }

  _createClass(I18NProvider, [{
    key: 'getChildContext',
    value: function getChildContext() {
      return {
        i18n: this.i18n,
        timeZone: this.timeZone,
        user: this.user,
        permission: this.permission,
        license: this.license
      };
    }
  }, {
    key: 'componentWillReceiveProps',
    value: function componentWillReceiveProps(nextProps) {
      var _this2 = this;

      if (this.props.i18n != nextProps.i18n || this.props.timeZone != nextProps.timeZone || this.props.user != nextProps.user || this.props.permission != nextProps.permission || this.props.license != nextProps.license && this.promise) {
        this.promise = new _Promise(function (res, rej) {
          _this2.resolve = res;
          _this2.reject = rej;
        }).then(function () {
          _this2.i18n = nextProps.i18n;
          _this2.timeZone = nextProps.timeZone;
          i18NProviderUtils.getI18NValue = getI18NValue(nextProps.i18n);
          i18NProviderUtils.userDateFormat = userDateFormat(i18NProviderUtils.getI18NValue, nextProps.timeZone);
          _this2.setState({ reRender: true }, function () {
            setTimeout(function () {
              return _this2.setState({ reRender: false });
            }, 1);
          });
          _this2.promise = null;
        }, function () {
          _this2.promise = null;
        });
        this.props.onChange(this.resolve, this.reject);
      }
    }
  }, {
    key: 'render',
    value: function render() {
      return this.state.reRender ? null : Children.only(this.props.children);
    }
  }]);

  return I18NProvider;
}(React.Component);

export default I18NProvider;


I18NProvider.defaultProps = {
  i18n: emptyObj,
  timeZone: '',
  user: null,
  permission: null,
  license: null
};
I18NProvider.childContextTypes = {
  i18n: PropTypes.object,
  timeZone: PropTypes.string,
  user: PropTypes.object,
  permission: PropTypes.object,
  license: PropTypes.object
};

export var ConnectI18NProvider = connect(function (state, props) {
  var i18n = {};
  var timeZone = '';
  if (typeof props.i18n == 'function') {
    i18n = props.i18n(state);
  } else if (_typeof(props.i18n) == 'object') {
    i18n = props.i18n;
  }
  if (typeof props.timeZone == 'function') {
    timeZone = props.timeZone(state);
  } else if (_typeof(props.timeZone) == 'object') {
    timeZone = props.timeZone;
  }
  return {
    i18n: i18n,
    timeZone: timeZone
  };
})(I18NProvider);

export var AppProvider = connect(function (state, props) {
  var i18n = null;
  var timeZone = '';
  var user = null;
  var permission = null;
  if (typeof props.i18n == 'function') {
    i18n = props.i18n(state);
  } else if (_typeof(props.i18n) == 'object') {
    i18n = props.i18n;
  }
  if (typeof props.timeZone == 'function') {
    timeZone = props.timeZone(state);
  } else if (_typeof(props.timeZone) == 'object') {
    timeZone = props.timeZone;
  }
  if (typeof props.user == 'function') {
    user = props.user(state);
  } else if (_typeof(props.user) == 'object') {
    user = props.user;
  }
  if (typeof props.permission == 'function') {
    permission = props.permission(state);
  } else if (_typeof(props.permission) == 'object') {
    permission = props.permission;
  }
  if (typeof props.license == 'function') {
    license = props.license(state);
  } else if (_typeof(props.license) == 'object') {
    license = props.license;
  }
  return {
    i18n: i18n,
    timeZone: timeZone,
    user: user,
    permission: permission,
    license: license
  };
})(I18NProvider);