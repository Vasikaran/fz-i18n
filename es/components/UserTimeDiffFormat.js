var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }();

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; }

import React, { PropTypes, Component } from 'react';
import DateTimeDiffFormat from './DateTimeDiffFormat';

var UserTimeDiffFormat = function (_Component) {
	_inherits(UserTimeDiffFormat, _Component);

	function UserTimeDiffFormat() {
		_classCallCheck(this, UserTimeDiffFormat);

		return _possibleConstructorReturn(this, (UserTimeDiffFormat.__proto__ || Object.getPrototypeOf(UserTimeDiffFormat)).apply(this, arguments));
	}

	_createClass(UserTimeDiffFormat, [{
		key: 'render',
		value: function render() {
			var currentTime = new Date();
			var currentTimeUTCString = currentTime.toUTCString();
			var timeZoneString = this.context.timeZone;
			var _props = this.props,
			    to = _props.to,
			    today = _props.today,
			    yesterday = _props.yesterday,
			    tomorrow = _props.tomorrow,
			    others = _props.others,
			    ago = _props.ago,
			    later = _props.later,
			    format = _props.format;

			return React.createElement(DateTimeDiffFormat, { from: currentTimeUTCString, fromTimeZone: timeZoneString, to: to,
				toTimeZone: timeZoneString, today: today, yesterday: yesterday, tomorrow: tomorrow,
				others: others, ago: ago, later: later, format: format });
		}
	}]);

	return UserTimeDiffFormat;
}(Component);

export default UserTimeDiffFormat;

UserTimeDiffFormat.contextTypes = {
	timeZone: PropTypes.string
};