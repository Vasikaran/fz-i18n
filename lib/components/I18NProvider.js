'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.AppProvider = exports.ConnectI18NProvider = exports.i18NProviderUtils = undefined;

var _typeof = typeof Symbol === "function" && typeof Symbol.iterator === "symbol" ? function (obj) { return typeof obj; } : function (obj) { return obj && typeof Symbol === "function" && obj.constructor === Symbol && obj !== Symbol.prototype ? "symbol" : typeof obj; };

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }();

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

var _reactRedux = require('react-redux');

var _utils = require('../utils');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; }

var emptyObj = {};
var dummy = function dummy(key, values) {
  console.log('Not Mount <I18NProvider/> component.');
  return key;
};
var i18NProviderUtils = exports.i18NProviderUtils = {
  getI18NValue: dummy,
  userDateFormat: dummy
};

var I18NProvider = function (_React$Component) {
  _inherits(I18NProvider, _React$Component);

  function I18NProvider(props, context) {
    _classCallCheck(this, I18NProvider);

    var _this = _possibleConstructorReturn(this, (I18NProvider.__proto__ || Object.getPrototypeOf(I18NProvider)).call(this, props, context));

    _this.i18n = props.i18n;
    _this.timeZone = props.timeZone;
    _this.user = props.user;
    _this.permission = props.permission;
    _this.license = props.license;
    _this.promise = null;
    _this.resolve = null;
    _this.reject = null;
    _this.state = { reRender: false };
    i18NProviderUtils.getI18NValue = (0, _utils.getI18NValue)(props.i18n);
    i18NProviderUtils.userDateFormat = (0, _utils.userDateFormat)(i18NProviderUtils.getI18NValue, props.timeZone);
    return _this;
  }

  _createClass(I18NProvider, [{
    key: 'getChildContext',
    value: function getChildContext() {
      return {
        i18n: this.i18n,
        timeZone: this.timeZone,
        user: this.user,
        permission: this.permission,
        license: this.license
      };
    }
  }, {
    key: 'componentWillReceiveProps',
    value: function componentWillReceiveProps(nextProps) {
      var _this2 = this;

      if (this.props.i18n != nextProps.i18n || this.props.timeZone != nextProps.timeZone || this.props.user != nextProps.user || this.props.permission != nextProps.permission || this.props.license != nextProps.license && this.promise) {
        this.promise = new Promise(function (res, rej) {
          _this2.resolve = res;
          _this2.reject = rej;
        }).then(function () {
          _this2.i18n = nextProps.i18n;
          _this2.timeZone = nextProps.timeZone;
          i18NProviderUtils.getI18NValue = (0, _utils.getI18NValue)(nextProps.i18n);
          i18NProviderUtils.userDateFormat = (0, _utils.userDateFormat)(i18NProviderUtils.getI18NValue, nextProps.timeZone);
          _this2.setState({ reRender: true }, function () {
            setTimeout(function () {
              return _this2.setState({ reRender: false });
            }, 1);
          });
          _this2.promise = null;
        }, function () {
          _this2.promise = null;
        });
        this.props.onChange(this.resolve, this.reject);
      }
    }
  }, {
    key: 'render',
    value: function render() {
      return this.state.reRender ? null : _react.Children.only(this.props.children);
    }
  }]);

  return I18NProvider;
}(_react2.default.Component);

exports.default = I18NProvider;


I18NProvider.defaultProps = {
  i18n: emptyObj,
  timeZone: '',
  user: null,
  permission: null,
  license: null
};
I18NProvider.childContextTypes = {
  i18n: _propTypes2.default.object,
  timeZone: _propTypes2.default.string,
  user: _propTypes2.default.object,
  permission: _propTypes2.default.object,
  license: _propTypes2.default.object
};

var ConnectI18NProvider = exports.ConnectI18NProvider = (0, _reactRedux.connect)(function (state, props) {
  var i18n = {};
  var timeZone = '';
  if (typeof props.i18n == 'function') {
    i18n = props.i18n(state);
  } else if (_typeof(props.i18n) == 'object') {
    i18n = props.i18n;
  }
  if (typeof props.timeZone == 'function') {
    timeZone = props.timeZone(state);
  } else if (_typeof(props.timeZone) == 'object') {
    timeZone = props.timeZone;
  }
  return {
    i18n: i18n,
    timeZone: timeZone
  };
})(I18NProvider);

var AppProvider = exports.AppProvider = (0, _reactRedux.connect)(function (state, props) {
  var i18n = null;
  var timeZone = '';
  var user = null;
  var permission = null;
  if (typeof props.i18n == 'function') {
    i18n = props.i18n(state);
  } else if (_typeof(props.i18n) == 'object') {
    i18n = props.i18n;
  }
  if (typeof props.timeZone == 'function') {
    timeZone = props.timeZone(state);
  } else if (_typeof(props.timeZone) == 'object') {
    timeZone = props.timeZone;
  }
  if (typeof props.user == 'function') {
    user = props.user(state);
  } else if (_typeof(props.user) == 'object') {
    user = props.user;
  }
  if (typeof props.permission == 'function') {
    permission = props.permission(state);
  } else if (_typeof(props.permission) == 'object') {
    permission = props.permission;
  }
  if (typeof props.license == 'function') {
    license = props.license(state);
  } else if (_typeof(props.license) == 'object') {
    license = props.license;
  }
  return {
    i18n: i18n,
    timeZone: timeZone,
    user: user,
    permission: permission,
    license: license
  };
})(I18NProvider);