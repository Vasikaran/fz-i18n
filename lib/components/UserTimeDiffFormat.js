'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }();

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _DateTimeDiffFormat = require('./DateTimeDiffFormat');

var _DateTimeDiffFormat2 = _interopRequireDefault(_DateTimeDiffFormat);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; }

var UserTimeDiffFormat = function (_Component) {
  _inherits(UserTimeDiffFormat, _Component);

  function UserTimeDiffFormat() {
    _classCallCheck(this, UserTimeDiffFormat);

    return _possibleConstructorReturn(this, (UserTimeDiffFormat.__proto__ || Object.getPrototypeOf(UserTimeDiffFormat)).apply(this, arguments));
  }

  _createClass(UserTimeDiffFormat, [{
    key: 'render',
    value: function render() {
      var currentTime = new Date();
      var currentTimeUTCString = currentTime.toUTCString();
      var timeZoneString = this.context.timeZone;

      var _props = this.props,
          to = _props.to,
          today = _props.today,
          yesterday = _props.yesterday,
          tomorrow = _props.tomorrow,
          others = _props.others,
          ago = _props.ago,
          later = _props.later,
          format = _props.format,
          title = _props.title,
          className = _props.className;

      return _react2.default.createElement(_DateTimeDiffFormat2.default, {
        from: currentTimeUTCString,
        fromTimeZone: timeZoneString,
        to: to,
        toTimeZone: timeZoneString,
        today: today,
        yesterday: yesterday,
        tomorrow: tomorrow,
        others: others,
        ago: ago,
        later: later,
        format: format,
        title: title,
        className: className
      });
    }
  }]);

  return UserTimeDiffFormat;
}(_react.Component);

exports.default = UserTimeDiffFormat;

UserTimeDiffFormat.contextTypes = {
  timeZone: _react.PropTypes.string
};