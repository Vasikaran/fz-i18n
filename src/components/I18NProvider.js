import React, { Children } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { getI18NValue, userDateFormat } from '../utils';

const emptyObj = {};
const dummy = (key, values) => {
  console.log('Not Mount <I18NProvider/> component.');
  return key;
};
export const i18NProviderUtils = {
  getI18NValue: dummy,
  userDateFormat: dummy
};
export default class I18NProvider extends React.Component {
  constructor(props, context) {
    super(props, context);
    this.i18n = props.i18n;
    this.timeZone = props.timeZone;
    this.user = props.user;
    this.permission = props.permission;
    this.license = props.license;
    this.promise = null;
    this.resolve = null;
    this.reject = null;
    this.state = { reRender: false };
    i18NProviderUtils.getI18NValue = getI18NValue(props.i18n);
    i18NProviderUtils.userDateFormat = userDateFormat(
      i18NProviderUtils.getI18NValue,
      props.timeZone
    );
  }
  getChildContext() {
    return {
      i18n: this.i18n,
      timeZone: this.timeZone,
      user: this.user,
      permission: this.permission,
      license: this.license
    };
  }
  componentWillReceiveProps(nextProps) {
    if (
      this.props.i18n != nextProps.i18n ||
      this.props.timeZone != nextProps.timeZone ||
      this.props.user != nextProps.user ||
      this.props.permission != nextProps.permission ||
      (this.props.license != nextProps.license && this.promise)
    ) {
      this.promise = new Promise((res, rej) => {
        this.resolve = res;
        this.reject = rej;
      }).then(
        () => {
          this.i18n = nextProps.i18n;
          this.timeZone = nextProps.timeZone;
          i18NProviderUtils.getI18NValue = getI18NValue(nextProps.i18n);
          i18NProviderUtils.userDateFormat = userDateFormat(
            i18NProviderUtils.getI18NValue,
            nextProps.timeZone
          );
          this.setState({ reRender: true }, () => {
            setTimeout(() => this.setState({ reRender: false }), 1);
          });
          this.promise = null;
        },
        () => {
          this.promise = null;
        }
      );
      this.props.onChange(this.resolve, this.reject);
    }
  }
  render() {
    return this.state.reRender ? null : Children.only(this.props.children);
  }
}

I18NProvider.defaultProps = {
  i18n: emptyObj,
  timeZone: '',
  user: null,
  permission: null,
  license: null
};
I18NProvider.childContextTypes = {
  i18n: PropTypes.object,
  timeZone: PropTypes.string,
  user: PropTypes.object,
  permission: PropTypes.object,
  license: PropTypes.object
};

export const ConnectI18NProvider = connect((state, props) => {
  var i18n = {};
  var timeZone = '';
  if (typeof props.i18n == 'function') {
    i18n = props.i18n(state);
  } else if (typeof props.i18n == 'object') {
    i18n = props.i18n;
  }
  if (typeof props.timeZone == 'function') {
    timeZone = props.timeZone(state);
  } else if (typeof props.timeZone == 'object') {
    timeZone = props.timeZone;
  }
  return {
    i18n,
    timeZone
  };
})(I18NProvider);

export const AppProvider = connect((state, props) => {
  var i18n = null;
  var timeZone = '';
  var user = null;
  var permission = null;
  if (typeof props.i18n == 'function') {
    i18n = props.i18n(state);
  } else if (typeof props.i18n == 'object') {
    i18n = props.i18n;
  }
  if (typeof props.timeZone == 'function') {
    timeZone = props.timeZone(state);
  } else if (typeof props.timeZone == 'object') {
    timeZone = props.timeZone;
  }
  if (typeof props.user == 'function') {
    user = props.user(state);
  } else if (typeof props.user == 'object') {
    user = props.user;
  }
  if (typeof props.permission == 'function') {
    permission = props.permission(state);
  } else if (typeof props.permission == 'object') {
    permission = props.permission;
  }
  if (typeof props.license == 'function') {
    license = props.license(state);
  } else if (typeof props.license == 'object') {
    license = props.license;
  }
  return {
    i18n,
    timeZone,
    user,
    permission,
    license
  };
})(I18NProvider);
