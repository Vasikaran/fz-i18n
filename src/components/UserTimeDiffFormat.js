import React, { PropTypes, Component } from 'react';
import DateTimeDiffFormat from './DateTimeDiffFormat';

export default class UserTimeDiffFormat extends Component{
	render(){
		var currentTime =  new Date();
		var currentTimeUTCString = currentTime.toUTCString();
		var timeZoneString = this.context.timeZone;
		var { to,today,yesterday,tomorrow,others,ago,later, format } = this.props;
		return <DateTimeDiffFormat from={currentTimeUTCString} fromTimeZone={timeZoneString} to={to} 
		toTimeZone={timeZoneString} today={today} yesterday={yesterday} tomorrow={tomorrow} 
		others={others} ago={ago} later={later} format={format}/>
	}
}
UserTimeDiffFormat.contextTypes = {
  timeZone: PropTypes.string
}