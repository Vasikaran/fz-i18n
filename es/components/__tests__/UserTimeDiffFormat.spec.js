import UserTimeDiffFormat from '../UserTimeDiffFormat';
import I18NProvider from '../I18NProvider';
import React from 'react';
import renderer from 'react-test-renderer';
var temp = Date;
describe('UserTimeDiffFormat component', function () {
  beforeAll(function () {
    Date = function Date(a) {
      if (a) {
        return new temp(a);
      }
      return new temp("2016-10-25T05:55:28.000Z");
    };
    Date.UTC = temp.UTC;
  });
  afterAll(function () {
    Date = temp;
  });
  it("Should display today key - user", function () {

    var ele = renderer.create(React.createElement(
      I18NProvider,
      { i18n: { today: "today", yesterday: "yesterday" }, timeZone: 'Asia/Kolkata' },
      React.createElement(UserTimeDiffFormat, { to: '2016-10-25T05:55:28.000Z',
        today: { key: "today", params: ["hh", "mm", "ss"] },
        yesterday: { key: "yesterday", params: ["hh", "mm", "ss"] },
        tomorrow: { key: "tomorrow", params: ["hh", "mm", "ss"] },
        others: function others(_ref) {
          var years = _ref.years,
              days = _ref.days,
              hours = _ref.hours,
              minutes = _ref.minutes;

          if (days > 7) {
            return "DD-MM-YYYY";
          } else {
            return getDateKeyWithParam(years, days, hours, minutes);
          }
        },
        ago: 'ago',
        later: 'later'
      })
    ));
    var tree = ele.toJSON();
    expect(tree).toMatchSnapshot();
  });
  it("Should display today later - user", function () {
    var ele = renderer.create(React.createElement(
      I18NProvider,
      { i18n: { today: "today", yesterday: "yesterday", "today.later": "{0}:{1}:{2} later" }, timeZone: 'Asia/Kolkata' },
      React.createElement(UserTimeDiffFormat, { to: '2016-10-25T05:55:29.000Z',
        today: { key: "today", params: ["hh", "mm", "ss"] },
        yesterday: { key: "yesterday", params: ["hh", "mm", "ss"] },
        tomorrow: { key: "tomorrow", params: ["hh", "mm", "ss"] },
        others: function others(_ref2) {
          var years = _ref2.years,
              days = _ref2.days,
              hours = _ref2.hours,
              minutes = _ref2.minutes;

          if (days > 7) {
            return "DD-MM-YYYY";
          } else {
            return getDateKeyWithParam(years, days, hours, minutes);
          }
        },
        ago: 'ago',
        later: 'later'
      })
    ));
    var tree = ele.toJSON();
    expect(tree).toMatchSnapshot();
  });
  it("Should display today ago - user", function () {
    Date = function Date(a) {
      if (a) {
        return new temp(a);
      }
      return new temp("2016-10-25T05:55:28.000Z");
    };
    Date.UTC = temp.UTC;
    var ele = renderer.create(React.createElement(
      I18NProvider,
      { i18n: { today: "today", yesterday: "yesterday", "today.later": "{0}:{1}:{2} later", "today.ago": "{0}:{1}:{2} ago" }, timeZone: 'Asia/Kolkata' },
      React.createElement(UserTimeDiffFormat, { to: '2016-10-25T05:55:27.000Z',
        today: { key: "today", params: ["hh", "mm", "ss"] },
        yesterday: { key: "yesterday", params: ["hh", "mm", "ss"] },
        tomorrow: { key: "tomorrow", params: ["hh", "mm", "ss"] },
        others: function others(_ref3) {
          var years = _ref3.years,
              days = _ref3.days,
              hours = _ref3.hours,
              minutes = _ref3.minutes;

          if (days > 7) {
            return "DD-MM-YYYY";
          } else {
            return getDateKeyWithParam(years, days, hours, minutes);
          }
        },
        ago: 'ago',
        later: 'later'
      })
    ));
    var tree = ele.toJSON();
    expect(tree).toMatchSnapshot();
  });
});