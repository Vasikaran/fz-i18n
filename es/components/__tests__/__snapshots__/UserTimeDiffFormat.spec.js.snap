exports[`UserTimeDiffFormat component Should display today ago - user 1`] = `
<span>
  00:00:01 ago
</span>
`;

exports[`UserTimeDiffFormat component Should display today key - user 1`] = `
<span>
  today
</span>
`;

exports[`UserTimeDiffFormat component Should display today later - user 1`] = `
<span>
  00:00:01 later
</span>
`;
